### DISCUSSION TITLE
Feature Request: Sub-filter for Data Provider

### DISCUSSION THREAD

**Discussion by chungchunwang** [Budibase/budibase#8148] [Ideas]
There are some situations in which you need to display data from the same table multiple times but filtered in different ways. Currently the only way to do this that I know of is to use multiple data providers each with a different filter setting. However, if too many of these are used, it really slows down the load time for the page. It would be nice to have a sub-filter component that is used as a child of Data Provider that can filter down the Data Provider data for its children. I have tried to make this into a plugin however I am unfamiliar with the SDK so I couldn't get the Provider component to work.

**Comment by melohagan** [comment]
Hey @chungchunwang 

You can select between *match all* and *match any* when filtering on a data provider: https://docs.budibase.com/docs/data-provider#filtering

Does that help in your case?

**Comment by chungchunwang** [comment]
Hi @melohagan , thanks for your input!
Not exactly. In my case I have a table of data in the internal Budibase DB which I wish to display in 6 separate tables each with a different filter. However, since filtering is tied with the Data Provider component, the page requires 6 separate Data Providers and thereby calls to the DB to load, which slows performance. 
My idea is that data that flows down from a single Data Provider can be then filtered into separate streams, which could then be fed into separate tables. In this case only one call to the DB would be necessary.
